app/graphql/types/query_type.rb

```diff
@@ -4,58 +4,34 @@ module Types
   class QueryType < Types::BaseObject
     description 'The query root of this schema'
 
-    field :user_id_query, [Types::EventType], null: false do
-      description 'returns all events with user_id'
-      argument :userId, ID, required: true
+    field :stats_count, [Types::EventType], null: false do
+      description 'returns counts of events in category grouped into period with length of interval (i.e. "1 day")'
+      argument :interval, String, required: true
+      argument :event_type, String, required: true
+      argument :user_id, ID, required: false
+      argument :project_id, ID, required: false
+      argument :workflow_id, ID, required: false
     end
 
-    def user_id_query(kwargs, searcher=Searchers::Complete)
-      if user_permission?(kwargs[:user_id])
-        searcher.new.search(**kwargs)
-      else
+    def stats_count(kwargs, searcher=Searchers::Bucket)
+      if site_wide_search?(kwargs) && (not context[:admin])
         raise GraphQL::ExecutionError, "Permission denied"
       end
-    end
-
-    field :project_id_query, [Types::EventType], null: false do
-      description 'returns all events with project_id'
-      argument :project_id, ID, required: true
-    end
-
-    def project_id_query(kwargs, searcher=Searchers::Complete)
-      searcher.new.search(**kwargs)
-    end
-
-    field :user_stats_count, [Types::EventType], null: false do
-      description 'returns counts of events by user in category grouped into period with length of interval (i.e. "1 day")'
-      argument :user_id, ID, required: true
-      argument :event_type, String, required: true
-      argument :interval, String, required: true
-    end
-
-    def user_stats_count(kwargs, searcher=Searchers::Bucket)
-      if user_permission?(kwargs[:user_id])
-        searcher.new.search(**kwargs)
+      if (not kwargs[:user_id]) || user_permission?(kwargs[:user_id])
```

Comment:
Maybe same here - again my understanding is allow filtering by user id in specific setups